src/ecs/systems/client/Draw.hpp

```diff
@@ -28,17 +29,35 @@ namespace ecs::systems
         auto const &positions = world.registry.getComponents<component::Position>();
         auto const &sizes = world.registry.getComponents<component::Size>();
         auto const &drawables = world.registry.getComponents<component::Drawable>();
+        auto &animateds = world.registry.getComponents<component::Animated>();
 
         world.getWindow().clear();
         for (size_t i = 0; i < positions.size() && i < sizes.size() && i < drawables.size(); ++i) {
             auto const &pos = positions[i];
             auto const &size = sizes[i];
             auto const &draw = drawables[i];
+
             if (pos && size && draw) {
                 sf::Sprite sprite;
+                float scaleX;
+                float scaleY;
                 sprite.setTexture(draw.value().Texture);
-                float scaleX = static_cast<float>(size.value().width) / static_cast<float>(draw.value().Texture.getSize().x);
-                float scaleY = static_cast<float>(size.value().height) / static_cast<float>(draw.value().Texture.getSize().y);
+                if (i < animateds.size() && animateds[i]) {
+                    auto &anim = animateds[i];
+                    sprite.setTextureRect(sf::IntRect(anim.value().width * anim.value().current + anim.value().origin_x, anim.value().origin_y, anim.value().width, draw.value().Texture.getSize().y));
+                    anim.value().cur_freq++;
+                    if (anim.value().cur_freq >= anim.value().freq) {
+                        anim.value().cur_freq = 0;
+                        anim.value().current++;
+                        if (anim.value().current * anim.value().width + anim.value().origin_x >= draw.value().Texture.getSize().x)
+                            anim.value().current = 0;
+                    }
+                    scaleX = static_cast<float>(size.value().width) / static_cast<float>(anim.value().width);
+                    scaleY = static_cast<float>(size.value().height) / static_cast<float>(draw.value().Texture.getSize().y);
+                } else {
+                    scaleX = static_cast<float>(size.value().width) / static_cast<float>(draw.value().Texture.getSize().x);
+                    scaleY = static_cast<float>(size.value().height) / static_cast<float>(draw.value().Texture.getSize().y);
```

Comment:
potential div. by zero.
```suggestion
                    if (static_cast<float>(draw.value().Texture.getSize().x != 0)
                        scaleX = static_cast<float>(size.value().width) / static_cast<float>(draw.value().Texture.getSize().x);
                    else
                        scaleX = 0;
                    if (static_cast<float>(draw.value().Texture.getSize().y != 0)
                        scaleY = static_cast<float>(size.value().height) / static_cast<float>(draw.value().Texture.getSize().y);
                    else
                        scaleY = 0;
```
or smth like that